core/src/test/scala/integration/kafka/api/ProducerFailureHandlingTest.scala

```diff
@@ -175,23 +176,23 @@ class ProducerFailureHandlingTest extends KafkaServerTestHarness {
   }
 
   /**
-   *  The send call with invalid partition id should throw KafkaException caused by IllegalArgumentException
-   */
+    * Send with invalid partition id should throw KafkaException when partition is higher than the
+    * upper bound of partitions and IllegalArgumentException when partition is negative
+    */
   @Test
   def testInvalidPartition() {
-    // create topic
+    // create topic with partition range [0...0]
```

Comment:
Didn't love to notation either, but that was what is printed in the error message. 

You think that [0...1) would be more clear? (Meaning that the upper bound is not inclusive)